refine model registry doc (#15882)
mlflow/mlflow: docs/docs/classic-ml/model-registry/index.mdx

---
sidebar_position: 15
toc_max_heading_level: 4
sidebar_label: Overview
pagination_next: model-registry/tutorial
---

import { APILink } from "@site/src/components/APILink";
import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

# MLflow Model Registry

> The MLflow Model Registry component is a centralized model store, set of APIs, and UI, to
> collaboratively manage the full lifecycle of an MLflow Model. It provides model lineage (which
> MLflow experiment and run produced the model), model versioning, model aliasing, model tagging, and
> annotations.

Reviewer: ```suggestion
The MLflow Model Registry is a centralized model store, set of APIs and a UI designed to
collaboratively manage the full lifecycle of a model. It provides lineage (i.e., which
MLflow experiment and run produced the model), versioning, aliasing, metadata tagging and
annotation support to ensure that you have the full spectrum of information at every stage from development to production deployment.
```

    </tr>
    <tr>
      <td>Registered Model</td>
      <td>
      An MLflow Model can be registered with the Model Registry. A registered model has a unique name, contains versions, aliases, tags, and other metadata.
      </td>
    </tr>
    <tr>
      <td>Model Version</td>
      <td>
      Each registered model can have one or many versions. When a new model is added to the Model Registry, it is added as version 1. Each new model registered to
      the same model name **increments the version number**. Model versions have tags, which can be useful for tracking attributes of the model version (e.g. _`pre_deploy_checks: "PASSED"`_)
      </td>
    </tr>
    <tr>
      <td>Model URI</td>
      <td>
      You can refer to the registered model by using a URI of this format: `models:/<model-name>/<model-version>`, e.g., if you have a registered model with name "MyModel" and version 1, the URI referring to the model is: `models:/MyModel/1`".
      </td>
    </tr>
    <tr>
      <td>Model Alias</td>
      <td>
      Model aliases allow you to assign a mutable, named reference to a particular version of a registered model. By assigning an alias to a specific model version,

>       you can use the alias to refer that model version via a model URI or the model registry API. For example, you can create an alias named **`champion`** that

Reviewer: ```suggestion
      you can use the alias to refer to that model version via a model URI or the model registry API. For example, you can create an alias named **`champion`** that
```

      Aliases are especially useful for deploying models. For example, you could assign a **`champion`** alias to the model version intended for production traffic
      and target this alias in production workloads. You can then update the model serving production traffic by reassigning the **`champion`** alias to a different model version.
      </td>
    </tr>
    <tr>
      <td>Tags</td>
      <td>
      Tags are key-value pairs that you associate with registered models and model versions, allowing you to label and categorize them by function or status. For example, you
      could apply a tag with key **`"task"`** and value **`"question-answering"`** (displayed in the UI as **`task:question-answering`**) to registered models intended for question
      answering tasks. At the model version level, you could tag versions undergoing pre-deployment validation with **`validation_status:pending`** and those cleared for deployment
      with **`validation_status:approved`**.
      </td>
    </tr>
    <tr>
      <td>Annotations and Descriptions</td>
      <td>

>       You can annotate the top-level model and each version individually using Markdown, including description and any relevant information useful for the team such as algorithm

Reviewer: ```suggestion
      You can annotate the top-level model and each version individually using Markdown, including the description and any relevant information useful for the team such as algorithm
```

<details>
  <summary>Register a model in MLflow</summary>

  <Tabs>
    <TabItem label="Python APIs" value="python-apis">
      #### Register a model with MLflow Python APIs

      MLflow provides several ways to register a model version
      ```
      # Option 1: specify `registered_model_name` parameter when logging a model
      mlflow.<flavor>.log_model(..., registered_model_name="<YOUR_MODEL_NAME>")

      # Option 2: register a logged model
      mlflow.register_model(model_uri="<YOUR_MODEL_URI>", name="<YOUR_MODEL_NAME>")
      ```

      After registering the model, you can load it back with the model name and version
      ```
      mlflow.<flavor>.load_model("models:/<YOUR_MODEL_NAME>/<YOUR_MODEL_VERSION>")
      ```

    </TabItem>
    <TabItem label="MLflow UI" value="mlflow-ui">
      #### Register a model on MLflow UI

      1. Open the details page for the MLflow Run containing the logged MLflow model you'd like to register. Select the model folder containing the intended MLflow model in the
        **Artifacts** section.

      ![](/images/oss_registry_1_register.png)

>       2. Click the **Register Model** button, which will trigger a form to pop up.

Reviewer: ```suggestion
      2. Click the **Register Model** button, which will trigger a modal form to pop up.
```

>       3. In the **Model** dropdown menu on the form, you can either select "Create New Model", which creates a new registered model with your MLflow model as its initial version,

Reviewer: ```suggestion
      3. In the **Model** dropdown menu on the form, you can either select "Create New Model" (which creates a new registered model with your MLflow model as its initial version)
```

>         or select an existing registered model, which registers your model under it as a new version. The screenshot below demonstrates registering the MLflow model to a new registered

Reviewer: ```suggestion
        or select an existing registered model (which registers your model under it as a new version). The screenshot below demonstrates registering the MLflow model to a new registered
```
